### PR TITLE
Nondeterministic Choice

### PR DESCRIPTION
**Pull request by CodaFi**
Adds the non-deterministic choice operator.

### CODE REVIEW

File: SwiftCheck/Operators.swift [hunk 1]
```diff
@@ -54,6 +54,23 @@ public func <?>(p : Testable, s : String) -> Property {
...
+	precedence 110
+}
+
+/// Takes the nondeterministic conjunction of two properties and treats them as a single large
+/// property.
+///
+/// The resulting property makes 100 random choices to test either the left or right property.  Thus,
+/// running multiple test cases will result in distinct arbitrary sequences of each property being
+/// tested.
+public func ^&^(p1 : Testable, p2 : Testable) -> Property {
```
Comment (robrix) [inline SwiftCheck/Operators.swift]: This is not going to be a very easy symbol to remember.

Comment (CodaFi) [inline SwiftCheck/Operators.swift]: This is QuickCheck's notation.  I couldn't find very many other historical examples of non-deterministic choice operators that looked good or weren't just `amb` hidden behind AoN.

Comment (robrix) [inline SwiftCheck/Operators.swift]: :ok_hand:

I really wish Swift supported Haskell’s backtick notation but they closed my radar as wontfix.

Comment (CodaFi) [inline SwiftCheck/Operators.swift]: Don't get me wrong, this gives me a bad feeling too, I just can't think of anything better.  Any ideas at all, I'll take em'.

Comment (robrix) [inline SwiftCheck/Operators.swift]: Well… `amb`. You don’t get nice infix syntax, but consumers can use `|>` or `<|` if they feel like it.

Comment (CodaFi) [inline SwiftCheck/Operators.swift]: Oh, so don't use an operator, just use `conjamb`?

Comment (CodaFi) [inline SwiftCheck/Operators.swift]: That's probably a better idea.  I can extend it to n argument then if we ever revisit the operator thing do what I did for conjoin and disjoin and make it a special 2-argument case.

Comment (robrix) [inline SwiftCheck/Operators.swift]: Yeah, exactly.

**Comment by CodaFi** [comment]
This does more good merged into the 2.0 branch.

**Pull request closed without merging**